examples/animation/animated_transform.rs

```diff
@@ -115,13 +115,13 @@ fn setup(
         ),
         VariableCurve {
             keyframe_timestamps: vec![0.0, 1.0, 2.0, 3.0, 4.0],
-            keyframes: Keyframes::Rotation(vec![
+            keyframes: Box::new(RotationKeyframes(vec![
```

Comment:
It is a shame we end up exposing the need for a box here. Might be nice to add a method(s) to `VariableCurve` to hide this detail? Perhaps something like:

```rust
pub fn linear(timestamps: Vec<f32>, keyframes: impl Keyframes) -> Self { ... }
```

Alternatively, since I imagine most `Keyframes` types will be simple wrappers around a `Vec`, perhaps we could do:

```rust
pub fn linear<K: Keyframes>(timestamps: Vec<f32>, keyframes: impl Into<K>) -> Self { ... }
```

Which could look nice when called:

```rust
VariableCurve::linear(
    vec![0.0, 1.0, 2.0],
    vec![1.0, 0.0, 1.0],
);
```

Reply:
There is also https://github.com/bevyengine/bevy/pull/13105 with a completely different API. But your suggestion is really nice and simple enough for now.

Reply:
Yeah, I was definitely aware of that PR. This API should be considered temporary until #13105 lands. Regardless, I like your suggestion.

Reply:
I added those constructors. Note that you do need a turbofish, so it doesn't quite look as nice as in your example.